docs/src/integrate/integration-tutorial.md

```diff
@@ -0,0 +1,252 @@
+
+- **Continuous Integration (CI) pipelines**: Including ESLint in CI pipelines can help enforce code quality standards before merging changes into the main codebase. By integrating ESLint into your CI process, you can automatically run linting checks on pull requests and provide feedback to developers about potential issues.
+
+- **Custom linter tools**: If you're building a custom linter tool that combines multiple linters or adds specific functionality, you may want to integrate ESLint into your tool to provide JavaScript linting capabilities.
+
+- **Code review tools**: Integrating ESLint with code review tools can help automate the process of identifying potential issues in the codebase.
+
+- **Learning platforms**: If you are developing a learning platform or coding tutorial, integrating ESLint can provide real-time feedback to users as they learn JavaScript, helping them improve their coding skills and learn best practices.
+
+## What You'll Build
+
+In this guide, you'll create a simple Node.js project that uses the `ESLint` class to lint files and retrieve results.
+
+## Requirements
+
+This tutorial assumes you are familiar with JavaScript and Node.js.
+
+To follow this tutorial, you'll need to have the following:
+
+- Node.js (v12.0.0 or higher)
+- npm or Yarn package manager
+- A text editor
+
+## Step 1: Setup
+
+First, create a new project directory:
+
+```shell
+mkdir eslint-integration
+cd eslint-integration
+```
+
+Initialize the project with a `package.json` file:
+
+```shell
+npm init -y
+```
+
+Install the `eslint` package as a dependency:
+
+```shell
+npm install eslint
+```
+
+Create a new file called `example-eslint-integration.js` in the project root:
+
+```shell
+touch example-eslint-integration.js
+```
+
+## Step 2: Import and Configure the `ESLint` Instance
+
+Import the `ESLint` class from the `eslint` package and create a new instance. 
+
+You can customize the ESLint configuration by passing an options object to the `ESLint` constructor:
+
+```javascript
+// example-eslint-integration.js
+
+const { ESLint } = require("eslint");
+
+function createESLintInstance(eslintConfig){
+    if (eslintConfig) {
+        return new ESLint({ useEslintrc: false, overrideConfig: eslintConfig });
+    } else {
+        return new ESLint({ useEslintrc: false });
+    }
+}
+```
+
+## Step 3: Get Linting Results
+
+To lint a file, use the `lintFiles` method of the `ESLint` instance:
+
+The `filePaths` argument passed to `ESLint#lintFiles()` can be a string or an array of strings, representing the file path(s) you want to lint.
+
+To format the linting results for better readability, use the `outputFixes` and `getErrorResults` methods:
```

Comment:
This text feels a bit disjointed to me and disconnected from the following code example. Can you expand this to talk a bit more about what each method does? For instance, lintFiles searches the filesystem and the file paths can be globs or filenames.

Also, outputFixes isn't for readability, it's the method that outputs fixes back to disk. getErrorResults filters out all warnings.

Reply:
updated copy here